python/PyQt6/core/auto_generated/project/qgsproject.sip.in

```diff
@@ -1343,7 +1343,7 @@ should use :py:func:`~QgsProject.addMapLayers` instead.
 .. seealso:: :py:func:`addMapLayers`
 %End
 
-    void removeMapLayers( const QStringList &layerIds );
+    void removeMapLayers( const QStringList &layerIds ) /PyName=removeMapLayersById/;
```

Comment:
Isn't this an API break? Maybe instead you'll need custom methodcode to check the list element types and then call the correct c++ method.

Reply:
is that ok if I add a patching in `python/PyQt6/core/__init__.py.in`?

Reply:
I'd personally prefer it was handled in method code, as that's compiled and we gain extra safety from it.

Reply:
I have reworked @benoitdm-oslandia's approach with a method code approach.